insights/parsers/sctp_eps.py

```diff
@@ -0,0 +1,109 @@
+"""
+SCTPEps - file ``/proc/net/sctp/eps``
+=====================================
+
+This provides detail information about SCTP endpoints.
+Which includes Endpoints, Socket, Socket type, Socket State,
+hash bucket, bind port, UID, socket inodes, Local IP
+address.
+
+Typical content of ``eps`` file is::
+
+    ENDPT            SOCK             STY SST HBKT LPORT   UID INODE     LADDRS
+    ffff88017e0a0200 ffff880300f7fa00 2   10  29   11165   200 299689357 10.0.0.102 10.0.0.70
+    ffff880612e81c00 ffff8803c28a1b00 2   10  30   11166   200 273361203 10.0.0.102 10.0.0.70 172.31.1.2
+
+Output data is stored in the dictionary format
+
+Examples:
+    >>> type(sctp_info)
+    <class 'insights.parsers.sctp_eps.SCTPEps'>
+    >>> sorted(sctp_info.get_ports()) == sorted(['11165', '11166'])
+    True
+    >>> sorted(sctp_info.get_local_ips()) == sorted(['10.0.0.102', '10.0.0.70', '172.31.1.2'])
+    True
+"""
+
+from insights import Parser, parser
+from insights.parsers import SkipException, ParseException
+from insights.specs import Specs
+
+
+@parser(Specs.sctp_eps)
+class SCTPEps(Parser):
+    """
+    This parser parses the content of ``/proc/net/sctp/eps`` file.
+    And returns a list of dictionaries. The dictionary contains details
+    of individual SCTP endpoint"
+    """
+
+    def parse_content(self, content):
+        if (not content) or (not self.file_path):
+            raise SkipException("No Contents")
+
+        COLUMN_IDX = {
+            'ENDPT': 'endpoints',
+            'SOCK': 'socket',
+            'STY': 'sk_type',
+            'SST': 'sk_state',
+            'HBKT': 'hash_bkt',
+            'LPORT': 'local_port',
+            'UID': 'uid',
+            'INODE': 'inode',
+            'LADDRS': 'local_addr'
+        }
+
+        self.data = []
```

Comment:
Have you considered using `parse_fixed_table` for this data? https://github.com/RedHatInsights/insights-core/blob/ac47281f9a83bb762982b14fb3b0824eb70d0c04/insights/parsers/__init__.py#L281

Reply:
@bfahr - We can not use this function as the number of columns are variable, Local address could be of one or more than one. Also, I have tested this functionality, it does not suit
Sample Input:
~~~
SCTP_EPS_DETAILS = """ 
 ENDPT     SOCK   STY SST HBKT LPORT   UID INODE LADDRS
ffff880422ab2600 ffff8804238c6ac0 2   10  28   3868    500 61868 10.0.200.114 10.0.201.114 2010:0010:0000:0200:0000:0000:0000:0114 2010:0010:0000:0201:0000:0000:0000:0114
""".STRIP()
~~~
Sample Output:
~~~
[{'ENDPT': 'ffff880422', 'SOCK': 'ab2600', 'STY': 'ffff', 'SST': '8804', 'HBKT': '238c6', 'LPORT': 'ac0 2', 'UID': '10', 'INODE': '28   3', 'LADDRS': '868    500 61868 10.0.200.114 10.0.201.114 2010:0010:0000:0200:0000:0000:0000:0114 2010:0010:0000:0201:0000:0000:0000:0114'}]
~~~

Reply:
That makes sense, thanks for checking.